src/main/java/io/cdap/plugin/servicenow/connector/ServiceNowConnector.java

```diff
@@ -0,0 +1,229 @@
+    }
+    return specBuilder.addRelatedPlugin(new PluginSpec(ServiceNowConstants.PLUGIN_NAME, BatchSource.PLUGIN_TYPE,
+        properties))
+      .addRelatedPlugin(new PluginSpec(ServiceNowConstants.PLUGIN_NAME, BatchSink.PLUGIN_TYPE, properties)).build();
+  }
+
+
+  private String getErrorMessage(String responseBody) {
+    try {
+      JsonObject jo = gson.fromJson(responseBody, JsonObject.class);
+      return jo.getAsJsonObject(ServiceNowConstants.ERROR).get(ServiceNowConstants.MESSAGE).getAsString();
+    } catch (Exception e) {
+      return e.getMessage();
+    }
+  }
+
+  @Override
+  public List<StructuredRecord> sample(ConnectorContext connectorContext, SampleRequest sampleRequest)
+    throws IOException {
+    String table = sampleRequest.getPath();
+    if (table == null) {
+      throw new IllegalArgumentException("Path should contain table name.");
+    }
+    try {
+      return getTableData(table, sampleRequest.getLimit());
+    } catch (OAuthProblemException | OAuthSystemException e) {
+      throw new IOException("Unable to fetch the data.");
+    }
+  }
+
+  private List<StructuredRecord> getTableData(String tableName, int limit) throws OAuthProblemException,
+    OAuthSystemException {
+    ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
+      config.getRestApiEndpoint(), tableName, false)
+      .setExcludeReferenceLink(true)
+      .setDisplayValue(SourceValueType.SHOW_DISPLAY_VALUE)
+      .setLimit(limit);
+    RestAPIResponse apiResponse = null;
+    ServiceNowTableAPIClientImpl serviceNowTableAPIClient = new ServiceNowTableAPIClientImpl(config);
+    String accessToken = serviceNowTableAPIClient.getAccessToken();
+    requestBuilder.setAuthHeader(accessToken);
+    requestBuilder.setResponseHeaders(ServiceNowConstants.HEADER_NAME_TOTAL_COUNT);
+    apiResponse = serviceNowTableAPIClient.executeGet(requestBuilder.build());
+    List<Map<String, Object>> result = serviceNowTableAPIClient.parseResponseToResultListOfMap
+      (apiResponse.getResponseBody());
+    List<StructuredRecord> recordList = new ArrayList<>();
+    Schema schema = getSchema(tableName);
+    if (schema != null) {
+      List<Schema.Field> tableFields = schema.getFields();
+      for (int i = 0; i < result.size(); i++) {
+        StructuredRecord.Builder recordBuilder = StructuredRecord.builder(schema);
+        for (Schema.Field field : tableFields) {
+          String fieldName = field.getName();
+          Object fieldValue = config.convertToValue(fieldName, field.getSchema(), result.get(i));
+          recordBuilder.set(fieldName, fieldValue);
+        }
+        StructuredRecord structuredRecord = recordBuilder.build();
+        recordList.add(structuredRecord);
+      }
+    }
+    return recordList;
+
+  }
+
+  @Nullable
+  private Schema getSchema(String tableName) {
+    SourceQueryMode mode = SourceQueryMode.TABLE;
+    SourceValueType valueType = SourceValueType.SHOW_DISPLAY_VALUE;
```

Comment:
Are we using `SHOW_ACTUAL_VALUE` anywhere? https://github.com/data-integrations/servicenow-plugins/blob/develop/src/main/java/io/cdap/plugin/servicenow/util/SourceValueType.java#L32

Reply:
No, I can't see any usage in the code.

Reply:
Should we remove it if it's unused?

Reply:
We should not, as Servicenow provides two types of Source Values. So it's better to always differentiate bw them that which type we are using.